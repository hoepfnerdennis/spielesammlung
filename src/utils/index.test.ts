import { chunkArray, mapResultsToGames, order, sortByName } from './index';
import data from '../../mockData/test-data.json';
import getGame from '../../mockData/test-games';

describe('utils', () => {
  describe('chunkArray', () => {
    it('should split array', () => {
      const input: number[] = [1, 2, 3, 4, 5];
      const output = chunkArray<number>(input, 3);
      expect(output).toHaveLength(2);
      expect(output[0]).toEqual([1, 2, 3]);
      expect(output[1]).toEqual([4, 5]);
    });
  });
  describe('order', () => {
    it('should return order', () => {
      expect(order(3, 2)).toEqual(1);
    });
  });
  describe('mapResultsToGames', () => {
    it('should map to products', () => {
      // @ts-ignore
      const output = mapResultsToGames(data);
      expect(output).toMatchSnapshot();
    });
  });
  describe('sortByName', () => {
    it.each([
      ['asc', getGame({ name: 'a' }), getGame({ name: 'b' }), -1],
      ['desc', getGame({ name: 'b' }), getGame({ name: 'a' }), 1],
      ['equal', getGame({ name: 'a' }), getGame({ name: 'a' }), 0],
    ])('sort %s', (_, game1, game2, result) => {
      expect(sortByName(game1, game2)).toBe(result);
    });
  });
});
