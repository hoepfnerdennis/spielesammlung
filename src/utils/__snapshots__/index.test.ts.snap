// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils mapResultsToGames should map to products 1`] = `
Array [
  Object {
    "age": "12",
    "description": "Fünf Würfel und ein Becher für jeden. Ein abschätzender Blick, eine schnelle Hochrechnung - wie viele Fünfer sind im Spiel? Die Wetten steigen, der Blutdruck auch. Bis einer die Nerven verliert und fordert: Hoch die Becher!",
    "duration": "30 Minuten",
    "favorite": true,
    "id": "3j9fYpYXTcn7oKFweKUujq",
    "image": "",
    "name": "Bluff",
    "playersFrom": 2,
    "playersTo": 6,
    "simpleRules": true,
  },
  Object {
    "age": "12",
    "description": "ohne fav und simple",
    "duration": "30 Minuten",
    "favorite": false,
    "id": "3j9fYpYXTcn7oKFweKUujq",
    "image": "",
    "name": "Bluff",
    "playersFrom": 2,
    "playersTo": 6,
    "simpleRules": false,
  },
  Object {
    "age": "8",
    "description": "Zuerst Winnetou in Bad Segeberg besuchen und dann in Heidelberg das größte Weinfass der Welt besichtigen? Oder erst zum Dom nach Köln? Die Spieler von „Deutschlandreise“ begeben sich auf eine spannende und informative Reise quer durch Deutschland.",
    "duration": "30 - 45 Minuten",
    "favorite": false,
    "id": "1UqzcWAO8OJwHvhKBhXKXW",
    "image": "//images.ctfassets.net/9sxha2f3gm24/5esCmTrGAPRhn0lW3SXyp2/8de3067ec9c3dc67c87c63e4abee9798/IMG_0213.jpg",
    "name": "Deutschlandreise",
    "playersFrom": 2,
    "playersTo": 2,
    "simpleRules": false,
  },
  Object {
    "age": "10",
    "description": "Mit dem Flieger nach Paris? Auf dem Wasserweg nach Barcelona? Oder doch den Zug nach Mailand? Auf jeden Fall sind die Spieler bei „Europareise“ mit Flugzeug, Bahn, Auto oder Schiff kreuz und quer durch ganz Europa unterwegs! Ziel ist es, möglichst viele Reiseziele in sechs unterschiedlichen Regionen zu besuchen.",
    "duration": "30 - 45 Minuten",
    "favorite": true,
    "id": "1gaoT4oQeWAZaeBKeiho0H",
    "image": "",
    "name": "Europareise",
    "playersFrom": 2,
    "playersTo": 6,
    "simpleRules": false,
  },
  Object {
    "age": "8",
    "description": "Mit reichlich Karotten und noch mehr Siegeswillen geht´s an den Start. Doch der Möhrenvorrat reicht nicht aus, um damit direkt ins Ziel zu kommen. Also müssen die Spieler sich immer wieder mal geschickt zurückfallen lassen, um neue Kraft zu tanken.",
    "duration": "30 - 45 Minuten",
    "favorite": false,
    "id": "73XWqmdaCC9N70waq68Q3X",
    "image": "",
    "name": "Hase und Igel",
    "playersFrom": 2,
    "playersTo": 6,
    "simpleRules": true,
  },
  Object {
    "age": "12",
    "description": "Im Koffer befinden sich zwei Kartensets mit je 52 Karten, Chips in rot (2x), weiß (2x), blau und schwarz, sowie einem Dealer, small blind und big blind.",
    "duration": "depends",
    "favorite": false,
    "id": "5tPtUeXotbR8qiWTgRJmn4",
    "image": "//images.ctfassets.net/9sxha2f3gm24/7mUfOSo47MiVzNUnHiAVg1/a1e0cfe06cdd5c1fa3b304e740c31d83/IMG_0214.png",
    "name": "Pokerkoffer",
    "playersFrom": 2,
    "playersTo": 10,
    "simpleRules": false,
  },
]
`;
